File: src/main/java/com/nixagh/classicmodels/entity/Customer.java
package com.nixagh.classicmodels.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "customers")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Customer implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customerNumber")
    private Long customerNumber;

    @Column(name = "customerName")
    private String customerName;

    @Column(name = "contactLastName")
    private String contactLastName;

    @Column(name = "contactFirstName")
    private String contactFirstName;

    @Column(name = "phone")
    private String phone;

    @Column(name = "addressLine1")
    private String addressLine1;

    @Column(name = "addressLine2")
    private String addressLine2;

    @Column(name = "city")
    private String city;

    @Column(name = "state")
    private String state;

    @Column(name = "postalCode")
    private String postalCode;

    @Column(name = "country")
    private String country;

    @ManyToOne
    @JoinColumn(name = "salesRepEmployeeNumber") // thông qua khóa ngoại
//	@JsonIgnore
    private Employee salesRepEmployeeNumber;

    @Column(name = "creditLimit")
    private Double creditLimit;

    @OneToMany(targetEntity = Order.class, mappedBy = "customer")
    @JsonIgnore
    private List<Order> ordersList = new ArrayList<>();

    @OneToMany(mappedBy = "customer")
    @JsonIgnore
    private List<Payment> payments = new ArrayList<>();

}

File: src/main/resources/static/js/customer.js
const loadCustomers = () => {
    callAPI.get(customerURL).then((response) => {
        let content = "";
        console.log(response);
        for (let i = 0; i < response.length; i++) {
            content += `
            <tr>
                <td>${response[i]["customerNumber"]}</td>
                <td>${response[i]["customerName"]}</td>
                <td>${response[i]["contactLastName"]}</td>
                <td>${response[i]["contactFirstName"]}</td>
                <td>${response[i]["phone"]}</td>
                <td>${response[i]["addressLine1"]}</td>
                <td>${response[i]["addressLine2"]}</td>
                <td>${response[i]["city"]}</td>
                <td>${response[i]["state"]}</td>
                <td>${response[i]["postalCode"]}</td>
                <td>${response[i]["country"]}</td>
                <td>${response[i]["salesRepEmployeeNumber"] == null ? null : response[i]["salesRepEmployeeNumber"]["employeeNumber"]}</td>
                <td>${response[i]["creditLimit"]}</td>
                <td>
                    <a href="#" class="btn btn-primary" onclick="getCustomer(${response[i]["customerNumber"]})">Edit</a>
                    <a href="#" class="btn btn-danger" onclick="deleteCustomer(${response[i]["customerNumber"]})">Delete</a>
                </td>
            </tr>
            `;
        }
        document.getElementById("customerList").innerHTML = content;
    })
}
loadCustomers();

const getCustomer = (id) => {

}

const deleteCustomer = (id) => {
    callAPI.delete(customerURL + "/" + id).then((response) => {
        alert("Delete customer successfully!");
    });
}